gateway/testutil.go

```diff
@@ -278,25 +429,25 @@ func firstVals(vals map[string][]string) map[string]string {
 	return m
 }
 
-type tykTestServerConfig struct {
+type MockConfig struct {
 	sepatateControlAPI bool
 	delay              time.Duration
 	hotReload          bool
 	overrideDefaults   bool
 	coprocessConfig    config.CoProcessConfig
 }
 
-type tykTestServer struct {
+type Mock struct {
```

Comment:
I think `Mock` is not the best name because here it doesn't mocking anything, in fact it runs real network listener with tyk